### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by Arkkienkeli**
All tests pass now, but probably will require more work in future.

### CODE REVIEW

File: deepprofiler/dataset/image_dataset.py [hunk 1]
```diff
@@ -29,12 +28,14 @@ def load_loc(self, params):
         # Load cell locations for one image
         i, config = params
         loc = deepprofiler.imaging.boxes.get_locations(self.keys[i], config)
+        print('locations', loc)
```
Comment (jccaicedo) [inline deepprofiler/dataset/image_dataset.py]: Remove these prints!

File: deepprofiler/dataset/image_dataset.py [hunk 2]
```diff
@@ -29,12 +28,14 @@ def load_loc(self, params):
...
         loc["ID"] = loc.index
         loc["ImageKey"] = self.keys[i]
         loc["ImagePaths"] = "#".join(self.images[i])
         loc["Target"] = self.targets[i][0]
         loc["Outlines"] = self.outlines[i]
         print("Image", i, ":", len(loc), "cells", end="\r")
+        print('locations', loc)
```
Comment (jccaicedo) [inline deepprofiler/dataset/image_dataset.py]: Remove prints

File: tests/deepprofiler/learning/test_profiling.py [hunk 3]
```diff
@@ -120,11 +120,12 @@ def profile(config, dataset):
     return deepprofiler.learning.profiling.Profile(config, dataset)
 
 
-def test_init(config, dataset):
-    prof = deepprofiler.learning.profiling.Profile(config, dataset)
+def test_init(config, dataset, locations):
+    metadata = deepprofiler.dataset.image_dataset.read_dataset(config)
+    prof = deepprofiler.learning.profiling.Profile(config, metadata)
```
Comment (jccaicedo) [inline tests/deepprofiler/learning/test_profiling.py]: Why was the dataset object not used here?

File: tests/plugins/crop_generators/test_autoencoder_crop_generator.py [hunk 4]
```diff
@@ -62,13 +61,29 @@ def metadata(out_dir, make_struct):
...
+    locations = pd.DataFrame({
+        "Nuclei_Location_Center_X": np.random.randint(0, 128, 10),
+        "Nuclei_Location_Center_Y": np.random.randint(0, 128, 10)
+    })
+    locations.to_csv(path, index=False)
+
 
 @pytest.fixture(scope="function")
-def dataset(metadata, out_dir, config, make_struct):
+def dataset(metadata, out_dir, config, make_struct, locations):
```
Comment (jccaicedo) [inline tests/plugins/crop_generators/test_autoencoder_crop_generator.py]: Locations doesn't seem to be used in this function.

File: tests/plugins/crop_generators/test_mixup_crop_generator.py [hunk 5]
```diff
@@ -56,13 +58,28 @@ def metadata(config, make_struct):
     meta.splitMetadata(train_rule, val_rule)
     return meta
 
+@pytest.fixture(scope="function")
```
Comment (jccaicedo) [inline tests/plugins/crop_generators/test_mixup_crop_generator.py]: This piece of code seems to be reused over and over. Can we generalize it somehow?

File: tests/plugins/crop_generators/test_autoencoder_crop_generator.py (hunk 4 above)
Comment (Arkkienkeli) [reply tests/plugins/crop_generators/test_autoencoder_crop_generator.py]: They have to be initilized in order to make ```dset.prepare_training_locations()``` call work.

File: tests/deepprofiler/learning/test_profiling.py (hunk 3 above)
Comment (Arkkienkeli) [reply tests/deepprofiler/learning/test_profiling.py]: Fixed.

File: deepprofiler/dataset/image_dataset.py (hunk 2 above)
Comment (Arkkienkeli) [reply deepprofiler/dataset/image_dataset.py]: Fixed.

File: deepprofiler/dataset/image_dataset.py (hunk 1 above)
Comment (Arkkienkeli) [reply deepprofiler/dataset/image_dataset.py]: Fixed.

File: tests/plugins/crop_generators/test_mixup_crop_generator.py (hunk 5 above)
Comment (Arkkienkeli) [reply tests/plugins/crop_generators/test_mixup_crop_generator.py]: Probably we could make shared fixtures, as ```locations``` is not the only one fixture we use over and over. https://stackoverflow.com/questions/46005796/reuse-pytest-fixtures

**Comment by Arkkienkeli** [comment]
@jccaicedo Could you look at the updates? 
Moved out fixtures. 
Also in ```test_mixup_crop_generator::test_start``` I changed assert, seems right to me, but not 100% sure. 
```test_crop_generator_build_augmentation_graph``` does not seem to test anything (```assert``` is missing at the end, this test does not pass, nevertheless it was like this) do we need this test at all?

**Pull request merged**